wifiphisher/pywifiphisher.py

```diff
@@ -569,52 +575,55 @@ def start(self):
         try:
             if args.internetinterface:
-               internet_interface = self.network_manager.set_internet_iface(args.internetinterface)
+                if (self.network_manager.is_interface_valid(args.internetinterface) and
+                        self.network_manager.is_interface_wired(args.internetinterface)):
+                    internet_interface = args.internetinterface
+                    self.network_manager.unblock_interface(internet_interface)
             if not args.nojamming:
                 if args.jamminginterface and args.apinterface:
-                    mon_iface = self.network_manager.get_jam_iface(
-                        args.jamminginterface)
-                    ap_iface = self.network_manager.get_ap_iface(args.apinterface)
+                    if self.network_manager.is_interface_valid(args.jamminginterface, "monitor"):
+                        mon_iface = args.jamminginterface
+                        self.network_manager.unblock_interface(mon_iface)
+                    if self.network_manager.is_interface_valid(args.apinterface, "AP"):
+                        ap_iface = args.apinterface
                 else:
-                    mon_iface, ap_iface = self.network_manager.find_interface_automatically()
-                self.network_manager.set_jam_iface(mon_iface.get_name())
-                self.network_manager.set_ap_iface(ap_iface.get_name())
+                    mon_iface, ap_iface = self.network_manager.get_interface_automatically()
+
                 # display selected interfaces to the user
                 print ("[{0}+{1}] Selecting {0}{2}{1} interface for the deauthentication "
                        "attack\n[{0}+{1}] Selecting {0}{3}{1} interface for creating the "
-                       "rogue Access Point").format(G, W, mon_iface.get_name(), ap_iface.get_name())
+                       "rogue Access Point").format(G, W, mon_iface, ap_iface)
             else:
                 if args.apinterface:
-                    ap_iface = self.network_manager.get_ap_iface(
-                        interface_name=args.apinterface)
+                    if self.network_manager.is_interface_valid(args.apinterface, "AP"):
+                        ap_iface = args.apinterface
                 else:
-                    ap_iface = self.network_manager.get_ap_iface()
+                    ap_iface = self.network_manager.get_interface(True, False)
                 mon_iface = ap_iface
-                self.network_manager.set_ap_iface(ap_iface.get_name())
+
                 print ("[{0}+{1}] Selecting {0}{2}{1} interface for creating the "
-                       "rogue Access Point").format(G, W, ap_iface.get_name())
+                       "rogue Access Point").format(G, W, ap_iface)
+
+            # make sure interfaces are not blocked
+            self.network_manager.unblock_interface(ap_iface)
+            self.network_manager.unblock_interface(mon_iface)
```

Comment:
In various places, we also need to make sure that the cards are not invalid. See my comment on #580.